lib/taskana-core/src/main/java/pro/taskana/impl/TaskServiceImpl.java

```diff
@@ -1394,14 +1394,14 @@ private void initAttachment(AttachmentImpl attachment, Task newTask) {
         }
     }
 
-    BulkOperationResults<String, Exception> classificationChanged(String taskId, String classificationId)
+    BulkOperationResults<String, Exception> refreshPriorityAndDueDate(String taskId, String modifiedClassificationId)
```

Comment:
Why do we need the classificationId in the signature? The method would make more sense to me if we would remove it.

Reply:
Good point - we don't need this parameter really. I'll remove it.

Reply:
Thanks god, I removed the classificationId parameter. In this way I found an error in method refreshPriorityAndDueDate(). It contained the 2 lines:

            Classification classification = classificationService.getClassification(modifiedClassificationId);
            task.setClassificationSummary(classification.asSummary());

This is an error because the task to modify might have a completely different classification and only be affected because one of its attachments had the updated classification.

Reply:
I think, the test coverage is fair because we modify a classification that is widely used. We set prio to 1000 and SL to P15D. The result:
37 tasks are directly affected because the use this classification directly. 36 from these 37 don't have attachments, therefore they get the new prio / SL from the classification. 1 from these 37 has also an attachment with SL P8D, therefore it gets a due date of planned = 8 working days.
4 tasks are affected only indirectly because only an attachment uses the modified classification.
Since these tasks use a classification with SL P1D, they end up with a due date of planned + 1 working day.
I double checked the expected results and found that in the previous version, the wrong results were expected. Now everything should be correct.
I added the check for correct prio / due date for the indirectly affected tasks to the testcase